### DISCUSSION TITLE
[destination-snowflake] configurable batch sizes and flush frequency

### DISCUSSION THREAD

**Discussion by atxnyc** [airbytehq/airbyte#35249] [Connector Questions] [answered]
### Topic

Make snowflake batch sizes and flush frequency configurable

### Relevant information

Issue [31741](https://github.com/airbytehq/airbyte/issues/31741) was opened to increase the size at which snowflake flushes from 10mb to 200mb

More frequent flushes are great for making sure that data gets to snowflake as quickly as possible; however, they significantly increase snowflake costs on slower sources. If we could configure the snowflake connector flush size and maximum time between flushes it would allow users to tailor the setting to prioritize performance vs cost depending on the use case.

**Comment by evantahler** [comment, marked as answer]
We aren't going to be working on this in the near future, so I'm closing this issue.  The confounding principle is [checkpointing](https://airbyte.com/blog/checkpointing) - Airbyte wants to commit data at least once every 15 minutes.  So the upper bound of batch-size is 200mb (snowflake's recommendation), and the lower bound is "however much data the stream gets in 15 min".

[a comment by an automated account is not shown]
